Client/Frontend/Browser/OpenInHelper.swift

```diff
@@ -24,38 +24,107 @@ struct OpenInViewUX {
 enum FileType : String {
     case PDF = "pdf"
     case PASS = "pkpass"
+    case HTML = "html"
+    case TXT = "txt"
+    case DEFAULT = "default"
+}
+
+enum MimeType: String {
+    case PDF = "application/pdf"
+    case PASS = "application/vnd.apple.pkpass"
+    case HTML = "text/html"
+    case TEXT = "text/plain"
+    case DEFAULT = "default"
 }
 
 protocol OpenInHelper {
-    var openInView: OpenInView { get }
-    static func canOpen(url: NSURL) -> Bool
+    init(url: NSURL)
+    init?(response: NSURLResponse)
+    var openInView: OpenInView? { get }
+
+    static func canOpenType(type: String) -> Bool
     func open()
 }
 
 struct OpenInHelperFactory {
-    static func helperForURL(url: NSURL) -> OpenInHelper? {
-        if OpenPdfInHelper.canOpen(url) {
+
+    static func helperForURL(url: NSURL, mimeType: String? = nil) -> OpenInHelper? {
+        guard let type = mimeType ?? url.pathExtension else {
+            return nil
+        }
+        if OpenPdfInHelper.canOpenType(type) {
             return OpenPdfInHelper(url: url)
-        } else if OpenPassBookHelper.canOpen(url) {
+        } else if OpenPassBookHelper.canOpenType(type) {
             return OpenPassBookHelper(url: url)
+        } else if ShareFileHelper.canOpenType(type) {
+            return ShareFileHelper(url: url)
         }
 
         return nil
     }
+
+    static func helperForResponse(response: NSURLResponse) -> OpenInHelper? {
+        guard let type = response.MIMEType ?? response.URL?.pathExtension else {
+            return nil
+        }
+        if OpenPdfInHelper.canOpenType(type) {
+            return OpenPdfInHelper(response: response)
+        } else if OpenPassBookHelper.canOpenType(type) {
+            return OpenPassBookHelper(response: response)
+        } else if ShareFileHelper.canOpenType(type) {
+            return ShareFileHelper(response: response)
+        }
+        return nil
+    }
+}
+
+class ShareFileHelper: NSObject, OpenInHelper {
+    lazy var openInView: OpenInView? = nil
```

Comment:
Nit: Using `lazy` here doesn't do anything since it will lazy load nil into this property and get setup later anyways.